connectors/sources/jira.py

```diff
@@ -0,0 +1,467 @@
+                "label": "Jira Cloud service account id",
+                "type": "str",
+            },
+            "api_token": {
+                "value": "abc#123",
+                "label": "Jira Cloud API token",
+                "type": "str",
+            },
+            "host_url": {
+                "value": "http://127.0.0.1:8080",
+                "label": "Jira host url",
+                "type": "str",
+            },
+            "ssl_disabled": {
+                "value": DEFAULT_SSL_DISABLED,
+                "label": "Disable SSL verification. True if SSL is disabled else False",
+                "type": "bool",
+            },
+            "ssl_ca": {
+                "value": "",
+                "label": "SSL certificate",
+                "type": "str",
+            },
+            "enable_content_extraction": {
+                "value": DEFAULT_CONTENT_EXTRACTION,
+                "label": "Enable content extraction (true/false)",
+                "type": "bool",
+            },
+            "retry_count": {
+                "value": RETRIES,
+                "label": "Maximum retries for failed requests",
+                "type": "int",
+            },
+            "concurrent_downloads": {
+                "value": 50,
+                "label": "Number of concurrent downloads for fetching attachment content",
+                "type": "int",
+            },
+        }
+
+    def tweak_bulk_options(self, options):
+        """Tweak bulk options as per concurrent downloads support by jira
+
+        Args:
+            options (dictionary): Config bulker options
+
+        Raises:
+            Exception: Invalid configured concurrent_downloads
+        """
+        if self.concurrent_downloads > MAX_CONCURRENT_DOWNLOADS:
+            raise Exception(
+                f"Configured concurrent downloads can't be set more than {MAX_CONCURRENT_DOWNLOADS}."
+            )
+        options["concurrent_downloads"] = self.concurrent_downloads
+
+    def _validate_configuration(self):
+        """Validates whether user input is empty or not for configuration fields
+
+        Raises:
+            Exception: Configured keys can't be empty
+        """
+        logger.info("Validating Jira Configuration...")
+        connection_fields = (
+            ["host_url", "service_account_id", "api_token"]
+            if self.is_cloud
+            else ["host_url", "username", "password"]
+        )
+
+        if empty_connection_fields := [
+            field for field in connection_fields if self.configuration[field] == ""
+        ]:
+            raise Exception(
+                f"Configured keys: {empty_connection_fields} can't be empty."
+            )
+
+        if not self.ssl_disabled and self.certificate == "":
+            raise Exception("When ssl_disabled is False then ssl_ca can't be empty.")
+
+    def _generate_session(self):
+        """Generates a new session"""
+        logger.debug("Creating a session...")
```

Comment:
More information could help - what's the session about? What is it responsible for? What can go wrong with session and be helpful to see in logs to trace problems?

Reply:
I'll update the docstring and logger to add more info about it.